### DISCUSSION TITLE
Automatically Disassemble All Identified Functions

### DISCUSSION THREAD

**Discussion by FUN-00000000** [NationalSecurityAgency/ghidra#3615] [Q&A] [answered]
Hello, I'm currently dealing with a unique disassembly style and was wondering if anyone had input on the best way to do something.

Basically:
1. I have a firmware blob that has no debug symbols
2. I have a separate file full of debug symbols, which I'm able to import via the ImportSymbolsScript.py script.
3. I now have dozens of defined functions under my Symbol Tree, but Ghidra's built-in analyzers don't appear to disassemble any of the functions. I've tried running auto-analysis, which doesn't actually disassemble anything, even though I would think each defined function symbol is a valid place to start disassembly/analysis.

For example, the symbols I import may say that the function "main()" occurs at the memory address 0x80000000. Because Ghidra knows a valid function exists at that memory address (per the imported symbols), I assumed Ghidra would automatically disassemble the memory at this address, but so far I've had to disassemble each function manually.

I was wondering, is there an easy/shorthand way to automatically disassemble at each defined function? So far, my process has been clicking on each of the function names and pressing "D" to force disassemble. Then going to the next one, hitting "D", etc. If there were a way to automate this process it would greatly speed up my reverse engineering process.

I figured this overall process may be a bit backwards, as usually you identify functions through the process of disassembling the rest of the binary, but in this case I have the function definitions imported before any actual disassembly has taken place. I figured this backwards nature may be why I'm not getting any automatic assistance from Ghidra's analyzers, which is why I've been manually pressing D on everything :sweat_smile: If anyone has input that could assist me, I would greatly appreciate it.

Thank you!

**Comment by astrelsky** [comment, marked as answer]
The way the importers achieve this is by creating a "1 byte function". The function should then automatically be disassembled when the "disassemble entry points" analyzer is run.

**Reply by FUN-00000000** [reply]
Hello,

While a number of 1-byte functions were created, they were unfortunately still not disassembled when the 'disassemble entry points' analyzer was run. Luckily, I've now determined why! It turns out that analyzer in particular has a 'Respect Execute Flag' option, and none of my memory blocks were marked as executable due to the way .s19 files import into Ghidra.

I made a hacky script to automatically disassemble every 1-byte function after posting this question, but simply unchecking that box, or instead setting my memory blocks to executable seems like a much more graceful solution.

Apologies for a dumb question, **tl;dr make sure the memory block that your functions are in is set to executable :)**
